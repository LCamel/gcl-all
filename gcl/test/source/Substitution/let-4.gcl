var y : Int 
{: 
    Q x = y
    P x y = x = y
:}

y := y + 1

{ P 3 (Q 1) }
