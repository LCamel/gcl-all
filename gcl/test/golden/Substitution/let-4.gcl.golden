4: P 3 (Q [ y \ y + 1 ] 1)
  P 3 (Q [ y \ y + 1 ] 1) ===> 3 = Q [ y \ y + 1 ] 1
    3: Q [ y \ y + 1 ] 1
      Q [ y \ y + 1 ] 1 ===> y + 1
        
5: P 3 (Q [ y \ y + 1 ] 1)
  P 3 (Q [ y \ y + 1 ] 1) ===> 3 = Q [ y \ y + 1 ] 1
    6: Q [ y \ y + 1 ] 1
      Q [ y \ y + 1 ] 1 ===> y + 1
        